Fix chart.js method signatures

burtek opened the pull request:
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://www.chartjs.org/docs/latest/developers/api.html#destroy and below
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

burtek commented:
PR's ready now, not a draft anymore. re-ping @anuti @FabienLavocat @KentarouTakeda @larrybahr @mernen @josefpaij @danmana @guillaume-ro-fr @archy-bold @braincore @frabnt @alexdor @mahnuh @Conrad777 @adripanico @wertzui @lekoaf @ElianCordoba

@RyanCavanaugh btw, shouldn't @typescript-bot be waiting until PR is marked as ready? It mentioned everyone before the draft was marked as ready (I created draft at first).

frabnt commented:
@burtek  Did you check if chart.js doc is actually in sync with the source code?

burtek commented:
Not really, I've just been irritated with `generateLegend()` and `stop()` typing and corrected two more methods based on docs. I've just checked all four and indeed one of them is not as doc says, so I've corrected that one. All four methods I'm changing here are in-sync with source code at the moment. Did not check anything else, though

frabnt commented on types/chart.js/index.d.ts:
```diff
@@ -33,7 +33,7 @@ declare class Chart {
     render: ({duration, lazy, easing}?: Chart.ChartRenderProps) => {};
     stop: () => this;
-    resize: () => this;
+    resize: () => void;
```
Why did you ignore the boolean param the function takes as input? https://github.com/chartjs/Chart.js/blob/cbace1cfe2c1b70ac26d1ee798eb5841fcd978c5/src/core/core.controller.js#L264

frabnt commented on types/chart.js/chart.js-tests.ts:
```diff
@@ -202,5 +202,5 @@ const chartWithScriptedOptions = new Chart(new CanvasRenderingContext2D(), {
 });
 
-chartWithScriptedOptions.stop().resize().clear().destroy();
+chartWithScriptedOptions.stop().clear().destroy();
```
Did you remove the call to `resize()` just because it's not chainable anymore?

frabnt commented on types/chart.js/chart.js-tests.ts:
```diff
@@ -201,3 +201,6 @@ const chartWithScriptedOptions = new Chart(new CanvasRenderingContext2D(), {
 });
+
+chartWithScriptedOptions.stop().clear().destroy();
+const legend: string = chartWithScriptedOptions.generateLegend();
```
What's the point of storing the result of `generateLegend()` call?

frabnt commented:
@burtek Please use meaningful commit messages.

burtek commented:
I'll just re-check and re-do everything, it will be easier that way
